internal/object/db_types.go

```diff
@@ -27,3 +30,55 @@ type ObjectRow struct {
 func (d *ObjectRow) TableName() string {
 	return "object"
 }
+
+// LoadFromDB loads objects from the database matching the given id
+// Returns error on database failure.
+func LoadFromDB(ctx context.Context, db *icingadb.DB, id types.Binary) (*Object, error) {
+	objectRow := &ObjectRow{ID: id}
+	err := db.QueryRowxContext(ctx, db.Rebind(db.BuildSelectStmt(objectRow, objectRow)+` WHERE "id" = ?`), objectRow.ID).StructScan(objectRow)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch object: %w", err)
+	}
+
+	tags := map[string]string{"host": objectRow.Host}
+	if objectRow.Service.Valid {
+		tags["service"] = objectRow.Service.String
+	}
+
+	metadata := make(map[int64]*SourceMetadata)
+	var sourceMetas []*SourceMetadata
+	err = db.SelectContext(ctx, &sourceMetas, db.Rebind(db.BuildSelectStmt(&SourceMetadata{}, &SourceMetadata{})+` WHERE "object_id" = ?`), id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch source object: %w", err)
+	}
+
+	var extraTags []*ExtraTagRow
+	err = db.SelectContext(
+		ctx, &extraTags,
+		db.Rebind(db.BuildSelectStmt(&ExtraTagRow{}, &ExtraTagRow{})+` WHERE "object_id" = ?`), id,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch object extra tags: %w", err)
+	}
+
+	for _, sourceMeta := range sourceMetas {
+		sourceMeta.ExtraTags = map[string]string{}
+		metadata[sourceMeta.SourceId] = sourceMeta
+	}
+
+	for _, extraTag := range extraTags {
+		sourceMeta, ok := metadata[extraTag.SourceId]
+		if ok {
+			sourceMeta.ExtraTags[extraTag.Tag] = extraTag.Value
+		}
+	}
+
+	obj := &Object{db: db, ID: id, Tags: tags, Metadata: metadata}
+
+	cacheMu.Lock()
+	defer cacheMu.Unlock()
+
+	cache[id.String()] = obj
+
+	return obj, nil
```

Comment:
We don't need such checks here, I don't even think that mutex lock is required, since this method is ever called when starting the daemon and before the listener becomes ready. And as discussed yesterday, an object can only have a single active incident at a time, i.e this method will never load the same object multiple times from the database.